### ISSUE TITLE
Cannot find 'windows.h' when compiling to hxcpp with MS Visual Studio 2013 Ultimate Installed

### ISSUE THREAD

**Issue by rknonchalance** [HaxeFoundation/hxcpp#492]
So I decided to install Microsoft Visual Studio 2013 Ultimate on my laptop to try and compile to windows through HXCPP. What's odd though is that it cannot find the header file for 'windows.h'. This is the error output:

```
Running process: S:\HaxeDevelop\Tools\fdbuild\fdbuild.exe "S:\Dev\WalangLaman\WalangLaman.hxproj" -ipc 0ce88edd-4fe0-4ac8-94e5-c9c4930ad6de -version "3.3.0" -compiler "S:\HaxeToolkit\haxe" -notrace -library "S:\HaxeDevelop\Library" -target ""
Building WalangLaman
Running Pre-Build Command Line...
cmd: haxe.exe -cp . -main Main -cpp yocpp
haxelib run hxcpp Build.xml haxe -Dhaxe3="1" -Dhaxe_ver="3.3" -Dhxcpp_api_level="330" -Dsource-header="Generated by Haxe 3.3.0" -I"./" -I"S:\\HaxeToolkit\\haxe\\extraLibs/" -I"" -I"S:\\HaxeToolkit\\haxe\\std/cpp/_std/" -I"S:\\HaxeToolkit\\haxe\\std/"
Creating S:/Dev/WalangLaman/yocpp/obj/msvc18xp/__pch/haxe/hxcpp.pch...
hxcpp.cpp
 - Compiling group 'haxe' with flags -Iinclude -nologo /WX- /fp:precise -DHX_WINDOWS -D_USING_V120_SDK71_ -GR -O2(optim-std) -Zi(debug) -FdS:\Dev\WalangLaman\yocpp\obj/msvc18xp/vc.pdb(debug) -Od(debug) -O2(release) -Os(optim-size) -FS -Oy- -c -EHs -GS- -arch:SSE -IS:/HaxeToolkit/haxe/lib/hxcpp/include -DHXCPP_VISIT_ALLOCS(haxe) -DHXCPP_API_LEVEL=330(haxe) -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -wd4996 tags=[haxe,static]
 - Compile : src/haxe/io/Eof.cpp
 - Compile : src/haxe/ds/StringMap.cpp
 - Compile : src/haxe/Log.cpp
 - Compile : src/haxe/IMap.cpp
 - Compile : src/Sys.cpp
 - Compile : src/Std.cpp
 - Compile : src/Print_Cross.cpp
 - Compile : src/Msg.cpp
 - Compile : src/Main.cpp
 - Compile : src/Exponent.cpp
 - Compile : src/__boot__.cpp
 - Compile : S:/HaxeToolkit/haxe/lib/hxcpp/src/hx/NoFiles.cpp
 - Compile : src/__resources__.cpp
 - Compiling group '__main__' with flags -Iinclude -DHX_DECLARE_MAIN -nologo /WX- /fp:precise -DHX_WINDOWS -D_USING_V120_SDK71_ -GR -O2(optim-std) -Zi(debug) -FdS:\Dev\WalangLaman\yocpp\obj/msvc18xp/vc.pdb(debug) -Od(debug) -O2(release) -Os(optim-size) -FS -Oy- -c -EHs -GS- -arch:SSE -IS:/HaxeToolkit/haxe/lib/hxcpp/include -DHXCPP_VISIT_ALLOCS(haxe) -DHXCPP_API_LEVEL=330(haxe) -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -wd4996 tags=[haxe,main,static]
 - Compile : src/__main__.cpp
Creating S:/Dev/WalangLaman/yocpp/obj/msvc18xp/__pch/runtime/hxcpp.pch...
hxcpp.cpp
 - Compiling group 'runtime' with flags -D_CRT_SECURE_NO_DEPRECATE -DHX_UNDEFINE_H -nologo /WX- /fp:precise -DHX_WINDOWS -D_USING_V120_SDK71_ -GR -O2(optim-std) -Zi(debug) -FdS:\Dev\WalangLaman\yocpp\obj/msvc18xp/vc.pdb(debug) -Od(debug) -O2(release) -Os(optim-size) -FS -Oy- -c -EHs -GS- -arch:SSE -IS:/HaxeToolkit/haxe/lib/hxcpp/include -DHXCPP_VISIT_ALLOCS(haxe) -DHXCPP_API_LEVEL=330(haxe) -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -wd4996 tags=[haxe]
 - Compile : src/hx/Anon.cpp
 - Compile : src/hx/Boot.cpp
 - Compile : src/hx/CFFI.cpp [haxe,static]
 - Compile : src/hx/Date.cpp
Error: While running :cl.exe -D_CRT_SECURE_NO_DEPRECATE -DHX_UNDEFINE_H -nologo /WX- /fp:precise -DHX_WINDOWS -D_USING_V120_SDK71_ -GR -O2 -FS -Oy- -c -EHs -GS- -arch:SSE -IS:/HaxeToolkit/haxe/lib/hxcpp/include -DHXCPP_VISIT_ALLOCS -DHXCPP_API_LEVEL=330 -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -wd4996 -MT -Yuhxcpp.h /FpS:/Dev/WalangLaman/yocpp/obj/msvc18xp/__pch/runtime/hxcpp.pch S:/HaxeToolkit/haxe/lib/hxcpp/src/hx/Date.cpp -FoS:/Dev/WalangLaman/yocpp/obj/msvc18xp/da3c452f_Date.obj
Date.cpp
S:/HaxeToolkit/haxe/lib/hxcpp/src/hx/Date.cpp(6) : fatal error C1083: Cannot open include file: 'windows.h': No such file or directory

Error: Build failed
Build halted with errors.
Done(1)
```
I'm using the latest version of hxcpp for this, and the ISO for the installer came directly from Microsoft. I've tried this last year before I reformatted the laptop due to a virus, and I had MS VS 2013 Professional installed, and although the compilation took a while, it worked last time. What could be the problem? I'm trying to search windows.h for the hopes that hxcpp is just looking for it at the wrong location like in the case of Android NDK.

**Comment by hughsando** [comment]
Windows.h comes separately in an winsdk - try https://developer.microsoft.com/en-us/windows/downloads/windows-10-sdk
The situation is slightly better on windows because the "vcvars.bat" file contains enough info to locate this stuff correctly.

**Comment by rknonchalance** [comment]
Thanks! I see. I'm using Windows 7 so will Windows 7 SDK suffice and/or will Windows 10 SKD be fine as well once it's installed? It's a bit strange that this didn't happen before last year - I was able to use hxcpp successfully and I never encountered the windows.h problem or the need for the SDK. I guess updates on the library and/or Haxe may have caused it?

**Comment by hughsando** [comment]
More likely to be related to a msvc upgrade - may have come pre-installed
with older versions, or the new version thinks I can't use an old sdk.
I think you are safe installing the latest windows sdk with an older
compiler.

**Comment by rknonchalance** [comment]
Good evening sir! I'm officially seriously embarrassed now. After installing Windows 7 SDK (which did not fix it yet), I realized the whole debacle was due to a simple name mismatch. The MSVC 2013 I currently had was Update 5 compared to last year's Update 4, and I found out under  Program Files (x86)/Microsoft SDKs/Windows that I already had it under v7.1A (instead of just 7.1A as written hxcpp's toolchain/msvc-setup.bat file). The Update changed the folder names apparently. Compiling works now.
